pact/policy-manager/policy-manager.pact

```diff
@@ -0,0 +1,339 @@
+        (if (exists-quote sale-id)
+          ;; quote is used
+          [ ;; Checks if update-quote-price exists
+            (if (exists-msg-decimal UPDATE-QUOTE-PRICE-MSG-KEY)
+              ;;true updates the quotes with the new price
+              (update-quote-price sale-id (read-decimal UPDATE-QUOTE-PRICE-MSG-KEY))
+              ;;false - skip
+              true
+            )
+            (map-escrowed-buy sale-id token seller buyer buyer-guard amount policies)
+          ]
+          ;; quote is not used
+          (map-buy token seller buyer buyer-guard amount sale-id policies)
+        )
+  )))
+
+  (defun enforce-transfer:[bool]
+    ( token:object{token-info}
+      sender:string
+      guard:guard
+      receiver:string
+      amount:decimal )
+    (enforce-ledger)
+    (with-capability (POLICY_MANAGER)
+      (let ((policies:[module{kip.token-policy-v2}]  (at 'policies token)))
+        (map-transfer token sender guard receiver amount policies))))
+
+
+;; Sale/Escrow Functions
+  (defun enforce-sale-pact:bool (sale:string)
+    "Enforces that SALE is id for currently executing pact"
+    (enforce (= sale (pact-id)) "Invalid pact/sale id")
+  )
+
+  (defun map-escrowed-buy:bool
+    ( sale-id:string
+      token:object{token-info}
+      seller:string
+      buyer:string
+      buyer-guard:guard
+      amount:decimal
+      policies:[module{kip.token-policy-v2}]
+    )
+    (let* (
+           (escrow-account:object{fungible-account} (get-escrow-account sale-id))
+           (quote:object{quote-schema} (get-quote-info sale-id))
+           (spec:object{quote-spec} (at 'spec quote))
+           (fungible:module{fungible-v2} (at 'fungible spec))
+           (buyer-fungible-account-name:string (read-msg BUYER-FUNGIBLE-ACCOUNT-MSG-KEY))
+           (seller-fungible-account:object{fungible-account} (at 'seller-fungible-account spec))
+           (price:decimal (at 'price spec))
+           (sale-price:decimal (floor (* price amount) (fungible::precision)))
+      )
+       ;; transfer fungible to escrow account
+       (fungible::transfer-create buyer-fungible-account-name (at 'account escrow-account) (at 'guard escrow-account) sale-price)
+
+       (with-capability (ESCROW sale-id)
+         ;; Run policies::enforce-buy
+         (map-buy token seller buyer buyer-guard amount sale-id policies)
+         ;; Transfer Escrow account to seller
+         (let (
+               (balance:decimal (fungible::get-balance (at 'account escrow-account)))
+             )
+             (install-capability (fungible::TRANSFER (at 'account escrow-account) (at 'account seller-fungible-account) balance))
+             (fungible::transfer (at 'account escrow-account) (at 'account seller-fungible-account) balance)
+         )
+       )
+       true
+    )
+  )
+
+  ;;utility functions
+
+  (defun exists-quote:bool (sale-id:string)
+    @doc "Looks up quote table for quote"
+    (= (take 6 (typeof (try false (get-quote-info sale-id)))) "object")
```

Comment:
For these kinds of `try`, you can simply use a let block with the potentially-failing code and return a boolean directly.

```pact
(try false (let ((q (get-quote-info sale-id))) true))
```